contrib/mpi-proxy-split/mpi-wrappers/p2p-deterministic.c

```diff
@@ -54,35 +54,38 @@ void initialize_next_msg(int fd) {
 }
 
 int iprobe_next_msg(struct p2p_log_msg *p2p_msg) {
-  /* FIXME:  This isn't comiling yet.
+  int rc = 0;
   if (!next_msg) {
-    initialize_next_msg(fd);
+    rc = get_next_msg(NULL);
+    if (rc == 1) return -1;
   }
-  */
   if (p2p_msg) {
     *p2p_msg = next_msg_entry;
   }
   return (next_msg_entry.comm != MPI_COMM_NULL);
 }
 
-void get_next_msg(struct p2p_log_msg *p2p_msg) {
-  static int fd = -2;
-  if (fd == -2) {
+int get_next_msg(struct p2p_log_msg *p2p_msg) {
+  static int fd = -1;
+  int rc = 1;
+  if (fd == -1) {
     char buf[100];
     int rank;
     MPI_Comm_rank(MPI_COMM_WORLD, &rank);
     snprintf(buf, sizeof(buf)-1, P2P_LOG_MSG, rank);
     fd = open(buf, O_RDONLY);
     if (fd == -1) {
-      perror("get_next_msg: open");
-      exit(1);
+      return 1;
```

Comment:
Could you add a comment here?
Normally, if we can't open the file, I would expect to give up.  But if we think we can continue, shouldn't we at least return `-1`, so that the caller can take some action?

Reply:
Yes. The first time we open this file is from REPLAY_PRE_IRECV(). At that time, this file has not created by LOG_POST_IRECV() yet. So the file does not exist and we expect it return "-1". We return `1` to tell caller there is no next message to replay. The caller will continue as if there is no next message to replay.